mkosi/config.py

```diff
@@ -2620,3 +2635,94 @@ def default(self, obj: Any) -> Any:
         elif isinstance(obj, uuid.UUID):
             return str(obj)
         return json.JSONEncoder.default(self, obj)
+
+
+E = TypeVar("E", bound=StrEnum)
+
+
+def json_type_transformer(refcls: Union[type[MkosiArgs], type[MkosiConfig]]) -> Callable[[str, Any], Any]:
+    fields_by_name = {field.name: field for field in dataclasses.fields(refcls)}
+
+    def path_transformer(path: str, fieldtype: type[Path]) -> Path:
+        return Path(path)
+
+    def optional_path_transformer(path: Optional[str], fieldtype: type[Optional[Path]]) -> Optional[Path]:
+        return Path(path) if path is not None else None
+
+    def path_list_transformer(pathlist: list[str], fieldtype: type[list[Path]]) -> list[Path]:
+        return [Path(p) for p in pathlist]
+
+    def optional_uuid_transformer(optuuid: Optional[str], fieldtype: type[Optional[uuid.UUID]]) -> Optional[uuid.UUID]:
+        return uuid.UUID(optuuid) if optuuid is not None else None
+
+    def root_password_transformer(
+        rootpw: Optional[list[Union[str, bool]]], fieldtype: type[Optional[tuple[str, bool]]]
+    ) -> Optional[tuple[str, bool]]:
+        if rootpw is None:
+            return None
+        return (cast(str, rootpw[0]), cast(bool, rootpw[1]))
+
+    def tree_transformer(
```

Comment:
I would call this source_target_transformer since not all trees support a target

Reply:
Changed.